### PR TITLE
basic rake task intergation via configurable buttons

### PR DESCRIPTION
**Pull request by DanHenton**
Acceptance Criteria
=================
- Add rake task support to plug.

Solution
=================

Checklist
=================
- [ ] [ ] Pull Request should have a descriptive title
- [ ] [ ] Code is understandable without Dev
- [ ] [ ] Acceptance criteria, what was changed, and why it was changed (If applicable) on Pull / Merge Request
- [ ] [ ] All new methods have a relevant spec

### CODE REVIEW

**Comment by DanHenton** [comment]
TODO: Update documentation on the addition of buttons + add some default panel images so that a visitor can see what plug looks like.

File: app/assets/stylesheets/plug/blocks/_task-button.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.task-button {
+  float: left;
```
Comment (hapiben) [inline app/assets/stylesheets/plug/blocks/_task-button.scss]: There's a helper class provided by `milligram CSS` to do float-left. You may need to wrap them into `clearfix` class as well. Same as `New feature` button.

File: app/views/plug/features/index.html.haml [hunk 2]
```diff
@@ -2,6 +2,13 @@
   .column
     - if notice
       %p.notice= notice
+    .row
```
Comment (hapiben) [inline app/views/plug/features/index.html.haml]: Move to a partial.

File: app/views/plug/features/index.html.haml [hunk 3]
```diff
@@ -2,6 +2,13 @@
...
+      .column
+        - Plug.buttons.each do |button|
+          .task-button
+            = form_tag('/plug/task') do
```
Comment (hapiben) [inline app/views/plug/features/index.html.haml]: Use `task_execution_path` instead.

File: config/routes.rb [hunk 4]
```diff
@@ -1,5 +1,6 @@
 Plug::Engine.routes.draw do
   root to: 'features#index'
 
+  post '/task', action: :task_execution, controller: :features, as: :task_execution
```
Comment (hapiben) [inline config/routes.rb]: `post '/task', to: 'features#task_execution', as: :task_execution`

File: spec/lib/plug/configuration_spec.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+RSpec.describe Plug::Configuration do
```
Comment (hapiben) [inline spec/lib/plug/configuration_spec.rb]: Space.

File: spec/lib/plug/configuration_spec.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+  let(:buttons_config) { [{ label: 'Clear Cache', task: 'tmp:cache:clear' }] }
+
+  it 'Allows the addition of buttons to the configuration' do
```
Comment (hapiben) [inline spec/lib/plug/configuration_spec.rb]: Use lowercase on all `it` blocks. Maybe rephrase this to be short. Something like:

 `it 'allows buttons'`

File: Gemfile.lock [hunk 7]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    plug (0.1.10)
+    plug (0.1.11)
```
Comment (hapiben) [inline Gemfile.lock]: Bump to `v0.1.13`.

File: app/views/plug/features/index.html.haml (hunk 3 above)
Comment (DanHenton) [reply app/views/plug/features/index.html.haml]: totally

File: Gemfile.lock (hunk 7 above)
Comment (DanHenton) [reply Gemfile.lock]: 👍

File: spec/controllers/plug/features_controller_spec.rb [hunk 8]
```diff
@@ -101,5 +101,26 @@ module Plug
       end
     end
 
+    describe 'POST #task_execution' do
+      let(:task) { 'tmp:cache:clear' }
+      it 'executes a rake task' do
```
Comment (hapiben) [inline spec/controllers/plug/features_controller_spec.rb:106]: Space

File: spec/lib/plug/configuration_spec.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Plug::Configuration do
+  let(:buttons_config) { [{ label: 'Clear Cache', task: 'tmp:cache:clear' }] }
+
+  it 'allows the addition of buttons to the configuration' do
```
Comment (hapiben) [inline spec/lib/plug/configuration_spec.rb:6]: No need to mention `to the configuration` because the test is describing the `Configuration` class.

**Review by hapiben — COMMENTED** [review]
Added some comments

**Review by tim-boost — APPROVED** [review]
